src/containers/registrationform.js

```diff
@@ -0,0 +1,162 @@
+            success: function(data) {
+              this.setState({fields: data});
+              console.log('success');
+            }.bind(this),
+            error: function(xhr, status, err) {
+              console.error(this.props.url, status, err.toString());
+              console.log('fail');
+            }.bind(this)
+          });
+
+    }
+
+    validateForm() {
+
+        let fields = this.state.fields;
+        let errors = {};
+        let formIsValid = true;
+
+        if (!fields["username"]) {
+            formIsValid = false;
+            errors["username"] = "*Please enter your username.";
+        }
+
+        if (typeof fields["username"] !== "undefined") {
+            if (!fields["username"].match(/^[a-zA-Z ]*$/)) {
+                formIsValid = false;
+                errors["username"] = "*Please enter alphabet characters only.";
+            }
+        }
+
+        if (!fields["emailid"]) {
+            formIsValid = false;
+            errors["emailid"] = "*Please enter your email-ID.";
+        }
+
+        if (typeof fields["emailid"] !== "undefined") {
+            //regular expression for email validation
+            var pattern = new RegExp(/^(("[\w-\s]+")|([\w-]+(?:\.[\w-]+)*)|("[\w-\s]+")([\w-]+(?:\.[\w-]+)*))(@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$)|(@\[?((25[0-5]\.|2[0-4][0-9]\.|1[0-9]{2}\.|[0-9]{1,2}\.))((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\.){2}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\]?$)/i);
+            if (!pattern.test(fields["emailid"])) {
+                formIsValid = false;
+                errors["emailid"] = "*Please enter valid email-ID.";
+            }
+        }
+
+        if (!fields["mobileno"]) {
+            formIsValid = false;
+            errors["mobileno"] = "*Please enter your mobile no.";
+        }
+
+        if (typeof fields["mobileno"] !== "undefined") {
+            if (!fields["mobileno"].match(/^[0-9]{10}$/)) {
+                formIsValid = false;
+                errors["mobileno"] = "*Please enter valid mobile no.";
+            }
+        }
+
+        if (!fields["password"]) {
+            formIsValid = false;
+            errors["password"] = "*Please enter your password.";
+        }
+
+        if (typeof fields["password"] !== "undefined") {
+            if (!fields["password"].match(/^.*(?=.{8,})(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%&]).*$/)) {
+                formIsValid = false;
+                errors["password"] = "*Please enter secure and strong password.";
+            }
+        }
+
+        this.setState({
+            errors: errors
+        });
+        return formIsValid;
+
+
+    }
+
+
+
+    render() {
+        return (
+            <div id="main-registration-container">
+                <div id="register">
+                    <h3>Registration page</h3>
+                    <form method="post" name="userRegistrationForm" onSubmit={this.submituserRegistrationForm} >
+                        <label>Name</label>
```

Comment:
Start converting each into single component,

Eg: A comonent which takes props and returns `label`, `input`, `div`. Group it into single `div`.

Reply:
Props passed can be, type of input, state value, name etc.